pwa/src/services/auth.tsx

```diff
@@ -20,4 +23,14 @@ export const isLoggedIn = () => {
 export const logout = () => {
   setUser({});
   window.sessionStorage.removeItem("jwt");
+  window.sessionStorage.removeItem("user");
 };
+
+export const validateSession = (jwt) => {
+  const decodedJwt = jwtDecode<JwtPayload>(jwt);
+
+  if (decodedJwt?.exp >= Date.now()) {
+    logout()
+    navigate("/login");
```

Comment:
I think navigating back to `/login` should not be part of a `validateSession` method, since it has nothing to do with validating the session itself. 

Even calling the `logout` method is debatable here. It might make more sense to implement one of the following two changes:

1. Let `validateSession` return a boolean (`true` for valid and `false` for invalid), when calling upon the `validateSession` method you can call the `logout` method depending on the outcome or
2. Move `navigate("/login")` to the `logout` method itself.